server/blob/blob-union/src/main/java/org/apache/james/blob/union/HybridBlobStore.java

```diff
@@ -0,0 +1,231 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.union;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.blob.api.ObjectNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+import reactor.core.publisher.Mono;
+
+public class HybridBlobStore implements BlobStore {
+    @FunctionalInterface
+    public interface RequireLowCost {
+        RequireHighPerformance lowCost(BlobStore blobStore);
+    }
+
+    @FunctionalInterface
+    public interface RequireHighPerformance {
+        RequireConfiguration highPerformance(BlobStore blobStore);
+    }
+
+    @FunctionalInterface
+    public interface RequireConfiguration {
+        Builder configuration(Configuration configuration);
+    }
+
+    public static class Builder {
+        private final BlobStore lowCostBlobStore;
+        private final BlobStore highPerformanceBlobStore;
+        private final Configuration configuration;
+
+        Builder(BlobStore lowCostBlobStore, BlobStore highPerformanceBlobStore, Configuration configuration) {
+            this.lowCostBlobStore = lowCostBlobStore;
+            this.highPerformanceBlobStore = highPerformanceBlobStore;
+            this.configuration = configuration;
+        }
+
+        public HybridBlobStore build() {
+            return new HybridBlobStore(
+                lowCostBlobStore,
+                highPerformanceBlobStore,
+                configuration);
+        }
+    }
+
+    public static class Configuration {
+        public static final int DEFAULT_SIZE_THREASHOLD = 32 * 1024;
+        public static final Configuration DEFAULT = new Configuration(DEFAULT_SIZE_THREASHOLD);
+        private static final String PROPERTY_NAME = "hybrid.size.threshold";
+
+        public static Configuration from(org.apache.commons.configuration2.Configuration propertiesConfiguration) {
+            return new Configuration(Optional.ofNullable(propertiesConfiguration.getInteger(PROPERTY_NAME, null))
+                .orElse(DEFAULT_SIZE_THREASHOLD));
```

Comment:
s/DEFAULT_SIZE_THREASHOLD/DEFAULT_SIZE_THRESHOLD

Reply:
I should have searched 'thea' in case insetive manner